### ISSUE TITLE
Publish 2.0.0 to npm

### ISSUE THREAD

**Issue by NaridaL** [mobxjs/serializr#125]
@mweststrate

**Comment by mweststrate** [comment]
Hey @NaridaL, just glanced through your recent PRs, and looks like you are doing some really solid work here! Awesome👏

If you are interested, I can give you publish rights on this package if you can share your npm user name. That means I won't be your bottleneck any longer ;-)

**Comment by NaridaL** [comment]
Works for me: https://www.npmjs.com/~narida

**Comment by mweststrate** [comment]
Done!

**Comment by NaridaL** [comment]
I've published 2.0.0-beta under @beta in case anyone else wants to test it too. Will publish 2.0.0 in a couple of days.

**Comment by mweststrate** [comment]
https://twitter.com/mweststrate/status/1226958538297290753

**Comment by danfma** [comment]
Good work, @NaridaL!

I'm using it on a project. I will do some tests today or tomorrow and then I give you feedback about it.

**Comment by danfma** [comment]
Hey, @NaridaL! 

It looks good until now.

But, before publishing, what do you think about exposing the context during the custom serialization in the serializer? With that, we could create a custom rule named `embedded` that will merge the properties of a composed child, directly in the parent serialization.

We could use this version increment to add that.

An example of that:

```typescript
class Name {
  @serializable(primitive())
  firstName = '';

  @serializable(primitive())
  lastName = '';
}

class Person {
  @serializable(embedded(Name))
  name = new Name();
}
```

So, a Person's instance could be serialized like:

```json
{
  firstName: 'name',
  lastName: 'surname'
}
```

Possible rule example:

```typescript
export function embedded<TModel>(type: Type<TModel>): PropSchema {
  return custom(
    (domain, context) => Object.assign(context.json, serialize(domain)),
    (_, context) => deserialize(type, context.json)
  );
}
```

**Comment by NaridaL** [comment]
@danfma There is currently no serialization context. However the afterDeserialize and beforeDeserialize callbacks get passed a "parentJson" object, maybe we could do the same for the serialization callback.

**Comment by danfma** [comment]
Yes, I saw that we have a context for the deserialization, so I thought we could have the same for the serialization (sorry, I didn't read the code).

But, if we have a {before/after}Serialization we could reach that too.

**Comment by NaridaL** [comment]
I've published 2.0.0. Closing this issue. @danfma Adding another parameter to the serialization function should be possible; you should open another issue or PR if you want to look into it further.
